File: Maryland-Cryptography/Programming_Assignment_3/proj3/sample.rb
require_relative "oracle"

if ARGV.length < 1
    puts "Usage: ruby sample.rb <filename>"
    exit
end

f = File.new(ARGV[0])
data = f.read
f.close

ctext = []
(0..data.length).step(2).each { |val|
    ctext.push(data[val,2].to_i(16))
}

Oracle_Connect()

rc = Oracle_Send(ctext, 3)
puts "Oracle returned: %d" % rc

Oracle_Disconnect()

File: Maryland-Cryptography/Programming_Assignment_3/proj3/sample.py
from oracle import *
import sys

if len(sys.argv) < 2:
    print "Usage: python sample.py <filename>"
    sys.exit(-1)

f = open(sys.argv[1])
data = f.read()
f.close()

ctext = [(int(data[i:i+2],16)) for i in range(0, len(data), 2)]

Oracle_Connect()

rc = Oracle_Send(ctext, 3)
print "Oracle returned: %d" % rc

Oracle_Disconnect()
